### DISCUSSION TITLE
Comparison to Storybook 7

### DISCUSSION THREAD

**Discussion by mvarendorff** [RyanClementsHax/storybook-addon-next#160] [General]
Storybook 7 is announced to be released with [zero config support for Next.js](https://storybook.js.org/blog/integrate-nextjs-and-storybook-automatically/) on March 14th with Storybook 7 currently being in beta. I haven't tested it yet but I was wondering how this project stacked up against it.

This is of course not to take away from the amazing work that has gone into the project! More than 100k downloads listed on the Storybook Integration store is no small feat.

**Comment by RyanClementsHax** [comment]
Thanks for the question; it's definitely a good one.

Their version of this addon (`@storybook/nextjs`) is actually something I helped develop. The plan is to deprecate this addon in favor of theirs since they can better support their users and the increasing amount of nextjs features. I plan on doing a few things to help this transition:
1. Creating migration documentation at the very least link to whatever storybook puts out
2. Supporting v7 with this addon
3. Maintaining this addon with only security patches for a few months after the release of v7 to help others transition

Side note: the team at storybook used this addon as the foundation for what they created with `@storybook/nextjs` so migration shouldn't be very painful

**Comment by jonniebigodes** [comment]
Hey @RyanClementsHax, I'm one of the Storybook maintainers. I focus primarily on documentation and community outreach. I'm reaching out to you to get a bit of information regarding your future plans for the addon and if you need any help from us with regards to any documentation changes or other changes that you might consider relevant, more even as you're already planning on including support for the upcoming 7.0 release as mentioned in your comment above. To help you with this, we've prepared an abridged guide for the required changes [here](https://chromatic-ui.notion.site/Addon-migration-for-Storybook-7-002f2f627c1d470298ca94894ab8a040), and if you have any questions or issues, please reach out to us in the #prerelease channel in our [Discord Server](https://discord.gg/storybook). 

Looking forward to hearing from you.

Hope you have a fantastic day.

Stay safe

**Reply by RyanClementsHax** [reply]
Thanks for reaching out. What bit of information were you looking to get that wasn't covered in the comment above?

Thanks for putting that doc together. Although there wasn't anything that applied to this addon, I'll keep it in mind to add what I see when I make the upgrade.

Please reach out again if you need anything from me. I've been spending more time on other repos recently, but won't be abandoning this one just yet. Also don't be afraid to be annoying with replies as you are competing for my attention with a 1 year old :)

**Reply by RyanClementsHax** [reply]
@jonniebigodes Actually there is one thing that would be helpful. What docs do you want me to point people to when they land on this repo? I hope to make things easier by redirecting people to where they can learn about migrating to storybook's frameworks.

**Reply by jonniebigodes** [reply]
Hey @RyanClementsHax 👋 
If you're ok with linking folks to this [recipe](https://storybook.js.org/recipes/next) we'd appreciate it. It's an abridged version of the full readme that should get folks onboarded as soon as possible. Also in it there's a link to the full Readme so that users can check it out in more detail. Also, if you spot any issues with this recipe, let us know and we'll be more than glad to make the required changes. Sounds good?

**Reply by RyanClementsHax** [reply]
Thanks. I currently don't see any issues with the documentation you provided.

I'll update the README of this repo to point to that. Lmk if any additional documentation comes out.

**Reply by jonniebigodes** [reply]
That's great to hear, and thanks for adding the change. Appreciate it.

**Comment by JReinhold** [comment]
Hi @RyanClementsHax!

Given you originally wrote this:
> Supporting v7 with this addon

Is that still your plan? If so, we've written a migration guide for addon authors [here](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#specific-instructions-for-addon-creators) and [here](https://chromatic-ui.notion.site/Addon-migration-for-Storybook-7-002f2f627c1d470298ca94894ab8a040)

Storybook v7.0.0 is out - on the `next` tag for now, but it's still targetable with the `^7.0.0` semver range - so you should be able to develop against it with no dependency issues.

If you have any further questions please reach out to us on Discord, you know where to find us! 👋

cc @ndelangen

**Reply by RyanClementsHax** [reply]
Oh! Thanks for reminding me! I got distracted with other projects I'm part of. I'll get this done soon.

**Reply by RyanClementsHax** [reply]
I partially added support for Storybook v7 by changing the addon's peer dependencies, but when I'm setting up an actual app with the addon and storybook v7, I'm running into odd webpack errors even after addressing the issues in the migration docs. The issue to track this problem is #186. I'm also looking in the storybook slack discord for the `#prerelease` channel to ask for help with this, but can't find one.

Is there a change with how config should be exported from addons?

**Reply by JReinhold** [reply]
> I'm also looking in the storybook slack discord for the #prerelease channel to ask for help with this, but can't find one.

That's odd, you don't see anything like this in Discord? 
![image](https://user-images.githubusercontent.com/5678122/229311722-93fd83a7-cf78-484e-a1ec-97a749144582.png)

**Reply by RyanClementsHax** [reply]
Nope all I see is the following. Do I need special permissions?

![image](https://user-images.githubusercontent.com/20916810/229314315-34e2efa5-fb6b-40d0-b187-1a694aa3b541.jpeg)

**Reply by RyanClementsHax** [reply]
I got it working. I didn't know I had to "opt in" to channels being shown to me.
